### DISCUSSION TITLE
Add Session ID getter method to Analytics

### DISCUSSION THREAD

**Discussion by giovaniortolani** [invertase/react-native-firebase#6677] [Ideas] [answered]
Hello!

Recently Firebase Analytics added a getter to retrieve the `ga_session_id` value.

This parameter is helpful if you are sending hits using Measurement Protocol V2 or doing user/session stitching between an app and web/webview.

Please consider updating and adding support to this getter in React Native Firebase as well.

Reference: https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics#public-tasklong-getsessionid (there is an implementation for iOS too but I'm linking the Android version only)

Last but not least. Probably in a near future, Firebase will add a getter for `ga_session_number` as this parameter is important to keep track of the session count.
So, keep in mind that too.

Reference: https://github.com/firebase/firebase-android-sdk/issues/3071#issue-1032227512

**Comment by mikehardy** [comment, marked as answer]
Hi there! In progress: #6633

**Reply by giovaniortolani** [reply]
Hi, @mikehardy, the PR was closed yesterday.

Is there any chance to reopen it? So another person can continue the development.

Thanks.

**Reply by mikehardy** [reply]
The PR state is irrelevant to anyone looking to continue the work, anyone may do so regardless of status by simply checking out that branch and doing whatever is necessary then posting a new PR. That's a point of git fact, it's not a policy statement or anything. So to any future readers, just clone the fork, checkout the branch and begin...
